### ISSUE TITLE
Product Report: Add Table

### ISSUE THREAD

**Issue by timmyc** [woocommerce/wc-admin#535]
Child task of #169 

**Design Post**: p6riRB-3dP-p2
**Mock API**: [https://app.swaggerhub.com/apis/peterfabian/wc-v3-api/1.0.0#/default/get_reports_products_stats](get_reports_products_stats)

Invision: [Large Viewport](https://automattic.invisionapp.com/d/main#/console/15060831/313168864/preview) | [Mobile](https://automattic.invisionapp.com/d/main#/console/15060831/313951190/preview)

This issue involves adding the table needed for the product report. The advanced filter will be added in another issue.

![image](https://user-images.githubusercontent.com/22080/46988447-88552080-d0ad-11e8-975b-558117380372.png)

**Comment by Aljullu** [comment]
I'm currently working on this but I'm not sure which approach should be used to populate the table with data.

To get all required data two API calls are needed:

* `/wc/v3/products` ([#](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products))
* `/wc/v3/reports/products` ([#](https://app.swaggerhub.com/apis/peterfabian/wc-v3-api/1.0.0#/default/get_reports_products))

The problem is that the first call doesn't allow ordering by `items_sold`, `gross_revenue` and `orders_count`. That means I can't get both queries to return the same products.

One option is requesting `/wc/v3/reports/products` first, and after that calling `/wc/v3/product` limiting the ids (with the `include` parameter) to the ones returned by the previous call. That would work but it means both calls can't be run in parallel.

@timmyc would that be a good implementation or is there some other way I am missing?

**Comment by timmyc** [comment]
@Aljullu if we use the [extended_product_info](https://github.com/woocommerce/wc-admin/blob/b653f1563dba2801bae5d986c2407873f2377d71/includes/api/class-wc-admin-rest-reports-products-controller.php#L245-L251) query arg on the `/wc/v3/reports/products` endpoint, does that get us the requisite product details needed for the table? It should get us the name, the only data it might not provide us is the category names.

**Comment by Aljullu** [comment]
Thanks @timmyc, I didn't know about that one!

However, even setting that argument to true, it doesn't return some of the values we need for the table: `sku`, `categories`, `variations`, `stock_status`, `stock_quantity`. Do you know if that is something that is going to be included in the future or we need to use both endpoints?

**Comment by timmyc** [comment]
Lets get as far as we can with what `extended_product_info` returns, and we can create a follow-up issue to look at included the other requisite fields in the API response. Not sure if all are do-able in a performant manner, but at least we can get as far as we can with the existing endpoint and iterate from there.

**Comment by Aljullu** [comment]
Sounds good, thanks @timmyc!
